syncthing/syncthing
# gui: Show a warning when authentication is not configured but listening on localhost (fixes #4703)

### Purpose

This shows a panel to warn the user that the UI can be accessed by other users on the same machine. It will only show when listening on localhost and remote access is not possible as there is already a warning for that.

The panel can be dismissed and it will not show up again across reloads and restarts.

I wasn't sure whether I should use the panel-warning or panel-info CSS class, can someone make a suggestion?

### Testing
- Set `STGUIADDRESS` to `127.0.0.1:8384` and disable auth -> "Warning!" panel appears
- Set `STGUIADDRESS` to `0.0.0.0:8384` and disable auth -> "Danger!" panel appears (can not be dismissed)
- Click `Do not show again` and reload page / restart Syncthing -> warning panel does not appear
- Set `STGUIADDRESS` to `127.0.0.1:8384` and setup auth -> warning panel does not appear

### Screenshots
![2019-10-27-000928_1187x618_scrot](https://user-images.githubusercontent.com/57039554/67626656-9906ad80-f84e-11e9-9450-ef429092e654.png)
